frontend/src/api/k8s/__tests__/dashboardConfig.spec.ts

```diff
@@ -0,0 +1,237 @@
+    });
+    expect(result).toStrictEqual(mockDashboardConfig({}));
+  });
+
+  it('should handle errors and rethrow', async () => {
+    k8sUpdateResourceMock.mockRejectedValue(new Error('error'));
+
+    await expect(updateDashboardConfig(mockDashboardConfig({}))).rejects.toThrow('error');
+    expect(k8sUpdateResourceMock).toHaveBeenCalledWith({
+      model: ODHDashboardConfigModel,
+      resource: mockDashboardConfig({}),
+    });
+    expect(k8sUpdateResourceMock).toHaveBeenCalledTimes(1);
+  });
+});
+
+describe('patchDashboardConfigTemplateOrder', () => {
+  const templateOrder = ['test - model'];
+
+  it('should patch dashboard config template order when template order is configured', async () => {
+    k8sPatchResourceMock.mockResolvedValue(mockDashboardConfig({}));
+
+    const result = await patchDashboardConfigTemplateOrder(templateOrder, projectName);
+    expect(k8sPatchResourceMock).toHaveBeenCalledTimes(1);
+    expect(k8sPatchResourceMock).toHaveBeenCalledWith({
+      model: ODHDashboardConfigModel,
+      queryOptions: { name: DASHBOARD_CONFIG, ns: projectName },
+      patches: [
+        {
+          op: 'replace',
+          path: '/spec/templateOrder',
+          value: templateOrder,
+        },
+      ],
+    });
+    expect(result).toStrictEqual(mockDashboardConfig({}).spec.templateOrder);
+  });
+
+  it('should throw error when template order is not configured', async () => {
+    const mockDashboardConfig = {
+      spec: {},
+    };
+    k8sPatchResourceMock.mockResolvedValue(mockDashboardConfig);
+
+    await expect(patchDashboardConfigTemplateOrder(templateOrder, projectName)).rejects.toThrow(
+      'Template order is not configured',
+    );
+    expect(k8sPatchResourceMock).toHaveBeenCalledTimes(1);
+    expect(k8sPatchResourceMock).toHaveBeenCalledWith({
+      model: ODHDashboardConfigModel,
+      queryOptions: { name: DASHBOARD_CONFIG, ns: projectName },
+      patches: [
+        {
+          op: 'replace',
+          path: '/spec/templateOrder',
+          value: templateOrder,
+        },
+      ],
+    });
+  });
+});
+
+describe('patchDashboardConfigTemplateDisablement', () => {
+  const templateDisablement = ['test - model'];
+
+  it('should patch dashboard config template disablement when template order is configured', async () => {
+    k8sPatchResourceMock.mockResolvedValue(mockDashboardConfig({}));
+
+    const result = await patchDashboardConfigTemplateDisablement(templateDisablement, projectName);
+    expect(k8sPatchResourceMock).toHaveBeenCalledTimes(1);
+    expect(k8sPatchResourceMock).toHaveBeenCalledWith({
+      model: ODHDashboardConfigModel,
+      queryOptions: { name: DASHBOARD_CONFIG, ns: projectName },
+      patches: [
+        {
+          op: 'replace',
+          path: '/spec/templateDisablement',
+          value: templateDisablement,
+        },
+      ],
+    });
+    expect(result).toStrictEqual(mockDashboardConfig({}).spec.templateDisablement);
+  });
```

Comment:
`const templateDisablement = ['test - model'];` is not used to create the mock value that's returned by the patch call. Although the test case is correctly asserting the values being used to call into the patch as well as the return value, it doesn't have a good flow. You should be creating the mock once, and using it throughout the test case. Similar comment can be applied to other tests in this file where a new mock is created multiple times per test but expecting them to be exactly the same values.

```suggestion
  it('should patch dashboard config template disablement when template order is configured', async () => {
    const mockDashboardConfig = mockDashboardConfig({});
    const { templateDisablement } = mockDashboardConfig.spec;
    k8sPatchResourceMock.mockResolvedValue(mockDashboardConfig);

    const result = await patchDashboardConfigTemplateDisablement(templateDisablement, projectName);
    expect(k8sPatchResourceMock).toHaveBeenCalledTimes(1);
    expect(k8sPatchResourceMock).toHaveBeenCalledWith({
      model: ODHDashboardConfigModel,
      queryOptions: { name: DASHBOARD_CONFIG, ns: projectName },
      patches: [
        {
          op: 'replace',
          path: '/spec/templateDisablement',
          value: templateDisablement,
        },
      ],
    });
    expect(result).toStrictEqual(templateDisablement);
  });
```

Reply:
const { templateDisablement } = mockDashboardConfig.spec here templateDisablement as type string[] | undefined which is not  compatible with the `patchDashboardConfigTemplateDisablement`argument type (string[])   
  Do you want me to do this instead 
  const { templateDisablement = [] } = dashboardConfigurationMock.spec;